File: src/main/java/com/hospital/management/prescription/service/PrescriptionService.java
package com.hospital.management.prescription.service;

import com.hospital.management.prescription.dto.PrescriptionRequestDTO;
import com.hospital.management.prescription.dto.PrescriptionResponseDTO;

import java.util.List;

public interface PrescriptionService {

    PrescriptionResponseDTO savePrescription(PrescriptionRequestDTO dto);

    List<PrescriptionResponseDTO> getAllPrescriptions();

    PrescriptionResponseDTO getPrescriptionById(Long id);

    PrescriptionResponseDTO updatePrescription(Long id, PrescriptionRequestDTO dto);

    void deletePrescription(Long id);
}

File: src/services/prescriptionService.js
import api from "./api";

export const getAllPrescriptions = () => {
  return api.get("/api/prescriptions");
};

export const savePrescription = (data) => {
  return api.post("/api/prescriptions", data);
};

export const deletePrescription = (id) => {
  return api.delete(`/api/prescriptions/${id}`);
};
export const updatePrescription = (id, data) => {
  return api.put(`/api/prescriptions/${id}`, data);
};